File: src/GoFit.Domain/Entities/WorkoutTracking.cs
using GoFit.Domain.Common;

namespace GoFit.Domain.Entities;

public class WorkoutTracking : BaseEntity
{
    public Guid WorkoutId { get; set; }
    public Workout Workout { get; set; }
    public Guid AthleteId { get; set; }
    public Athlete Athlete { get; set; }
    public DateTime StartWorkoutDate { get; set; }
    public DateTime? EndWorkoutDate { get; set; }
    public DateTime? CancelledDate { get; set; }
    public string? Note { get; set; }
    public ICollection<WorkoutSetTracking> Sets { get; set; }
}

File: src/client/src/app/shared/models/workout-tracking.ts
import { Workout } from './workout-plan';

export type WorkoutSetTracking = {
    repetitions: number;
    weight: number;
    order: number;
}

export type WorkoutTracking = {
    id: string;
    workoutId: string;
    workout: Workout;
    startWorkoutDate: string;
    endWorkoutDate?: string | null;
    cancelledDate?: string | null;
    note?: string | null;
    sets: WorkoutSetTracking[];
}

export type StartWorkoutTrackingRequest = {
    workoutId: string;
    note?: string | null;
    sets: WorkoutSetTracking[];
}

export type UpdateWorkoutTrackingRequest = {
    workoutsTrackingId: string;
    startWorkoutDate: string;
    endWorkoutDate?: string | null;
    note?: string | null;
    sets: WorkoutSetTracking[];
}
